dataless-classifier/src/main/java/edu/illinois/cs/cogcomp/datalessclassification/representation/AEmbedding.java

```diff
@@ -0,0 +1,173 @@
+
+    /**
+     * Override this function to get your desired query segmentation (simpleSegmentation being one
+     * example)
+     */
+    public List<String> getTerms(String query) {
+        return simpleSegmentation(query);
+    }
+
+    final private List<String> simpleSegmentation(String query) {
+        query = QueryPreProcessor.process(query);
+        String[] terms = query.split("\\s+");
+
+        List<String> termList = Arrays.asList(terms);
+
+        return termList;
+    }
+
+    /**
+     * Override this function to get your desired term processing module (simpleProcessTerm being
+     * one example)
+     */
+    public String processTerm(String s) {
+        return simpleProcessTerm(s);
+    }
+
+    final private String simpleProcessTerm(String s) {
+        return s.toLowerCase().trim();
+    }
+
+    /**
+     * Implement this function for returning the default vector in case no word in the query could
+     * be found in the index
+     * 
+     * @throws java.io.IOException
+     */
+    abstract public SparseVector<T> getDefaultConceptVectorMap();
+
+    /**
+     * Intended to output the vector for the most basic unit Should internally process the term
+     * using the processTerm function
+     * 
+     * @throws java.io.IOException
+     */
+    abstract public SparseVector<T> getTermConceptVectorMap(String term);
+
+    /**
+     * Intended to output the vector for any arbitrary query
+     * 
+     * @throws java.io.IOException
+     */
+    abstract public SparseVector<T> getVector(String query);
+
+    /**
+     * Override this function to select given number of dimensions
+     */
+    public SparseVector<T> getVector(String query, int numConcepts) {
+        return getVectorIgnoreSize(query);
+    }
+
+    private final SparseVector<T> getVectorIgnoreSize(String query) {
+        return getVector(query);
+    }
+
+    public SparseVector<T> getConceptVectorBasedonSegmentation(String query) {
+        return getConceptVectorBasedonSegmentation(query, false);
+    }
+
+    /**
+     * Switching off ignoreTermFreq will return a simple averaging over all the terms found in the
+     * index
+     * 
+     * @throws java.io.IOException
+     */
+    public SparseVector<T> getConceptVectorBasedonSegmentation(String query, boolean ignoreTermFreq) {
+        Map<String, Double> termWeights = new HashMap<>();
+        List<String> terms = getTerms(query);
+
+        for (String term : terms) {
+            if (termWeights.containsKey(term) == false) {
+                termWeights.put(term, 1.0);
+            } else {
+                if (ignoreTermFreq == false)
+                    termWeights.put(term, termWeights.get(term) + 1.0);
+                else
+                    continue;
+            }
+        }
+
+        return getConceptVectorBasedonTermWeights(termWeights);
+    }
+
+    public SparseVector<T> getConceptVectorBasedonTermWeights(Map<String, Double> termWeights) {
```

Comment:
javadoc explaining distinction with 'basedonSegmentation'
nitpick: capitalize the 'O' in 'basedOn'

Reply:
Done.